### PR TITLE
[stable/sumologic-fluentd] Add daemon set 1.6 features

### PR DESCRIPTION
**Pull request by flah00**
@frankreno

* podAnnotations, tolerations, and updateStrategy

### CODE REVIEW

**Review by frankreno — COMMENTED** [review]
LGTM

File: stable/sumologic-fluentd/templates/daemonset.yaml [hunk 1]
```diff
@@ -13,6 +13,10 @@ spec:
       name: {{ template "fullname" . }}
       labels:
         app: {{ template "fullname" . }}
+      {{- if .Values.daemonset.podAnnotations }}
```
Comment (mgoodness) [inline stable/sumologic-fluentd/templates/daemonset.yaml]: Nit: the `if` and `else` statements should be indented only one level less than the block they wrap.

File: stable/sumologic-fluentd/templates/daemonset.yaml [hunk 2]
```diff
@@ -117,3 +121,9 @@ spec:
         - name: docker-logs
           hostPath:
             path: /var/lib/docker
+      {{- if .Values.daemonset.tolerations }}
```
Comment (mgoodness) [inline stable/sumologic-fluentd/templates/daemonset.yaml]: Nit: the `if` and `else` statements should be indented only one level less than the block they wrap.

File: stable/sumologic-fluentd/templates/daemonset.yaml [hunk 3]
```diff
@@ -117,3 +121,9 @@ spec:
...
+      tolerations:
+{{ toYaml .Values.daemonset.tolerations | indent 8 }}
+      {{- end }}
+  updateStrategy:
+    type: {{ default "OnDelete" .Values.daemonset.updateStrategy | quote }}
```
Comment (mgoodness) [inline stable/sumologic-fluentd/templates/daemonset.yaml]: See comment on `values.yaml`, then `type: "{{ .Values.updateStrategy }}"` is sufficient

File: stable/sumologic-fluentd/values.yaml [hunk 4]
```diff
@@ -4,6 +4,21 @@ image:
   tag: latest
   pullPolicy: Always
 
+daemonset:
```
Comment (mgoodness) [inline stable/sumologic-fluentd/values.yaml]: Is anything gained by creating an intermediate `daemonset` object? Recommend making these top-level values.

File: stable/sumologic-fluentd/values.yaml [hunk 5]
```diff
@@ -4,6 +4,21 @@ image:
...
+  ## Annotations to add to the DaemonSet's Pods
+  podAnnotations:
```
Comment (mgoodness) [inline stable/sumologic-fluentd/values.yaml]: `podAnnotations: {}`

File: stable/sumologic-fluentd/values.yaml [hunk 6]
```diff
@@ -4,6 +4,21 @@ image:
...
+  #  scheduler.alpha.kubernetes.io/tolerations: '[{"key": "example", "value": "foo"}]'
+
+  ## Allow the DaemonSet to schedule on tainted nodes (requires Kubernetes >= 1.6)
+  # tolerations:
```
Comment (mgoodness) [inline stable/sumologic-fluentd/values.yaml]: Uncomment and make `tolerations: []`

File: stable/sumologic-fluentd/values.yaml [hunk 7]
```diff
@@ -4,6 +4,21 @@ image:
...
+  #   - key: node.alpha.kubernetes.io/role
+  #     effect: NoSchedule
+  #     operator: "Exists"
+
+  ## Allow the DaemonSet to perform a rolling update on helm update
+  ## ref: https://kubernetes.io/docs/tasks/manage-daemon/update-daemon-set/
+  # updateStrategy: RollingUpdate
```
Comment (mgoodness) [inline stable/sumologic-fluentd/values.yaml]: Uncomment and make `updateStrategy: OnDelete`

**Review by mgoodness — CHANGES_REQUESTED** [review]

**Comment by flah00** [comment]
@mgoodness thanks for the review, should be good to go

**Review by mgoodness — APPROVED** [review]

**Pull request merged**